src/js/step.js

```diff
@@ -309,7 +310,7 @@ export class Step extends Evented {
     let tourOptions =
       this.tour && this.tour.options && this.tour.options.defaultStepOptions;
 
-    tourOptions = tourOptions ? JSON.parse(JSON.stringify(tourOptions)) : {};
+    tourOptions = merge({}, tourOptions || {});
```

Comment:
@chuckcarpenter this is essentially the only logic change, switching `JSON.stringify` to use `deepmerge` instead so we get a deep clone.